### PR TITLE
Decouple contact form into a jetpack/forms package

### PR DESCRIPTION
**Pull request by ice9js**
## Proposed changes:

This PR adds a new jetpack/forms package and moves the existing contact form code to it.

As of right now, I've moved most if not all of the PHP/backend code, with form Gutenberg blocks the only thing remaining but we'll need to work on the setup for compiling them first.

I went ahead and introduced the following structure, that I think might make sense going forward:

```
packages/forms/src
    ├- blocks          # Assuming we decouple our blocks from the contact form implementation, that'd go here
    ├- block-editor    # Gutenberg blocks
    ├- contact-form    # Existing contact form implementation
    └- dashboard       # New feedback dashboard implementation
```

I took the opportunity to split up the contact form into code separate classes/files to make it a little easier to find stuff.

**To do's:**

- [ ] Exctract contact form gutenberg blocks from `plugins/jetpack`
- [ ] Add tooling for building a js bundle for the jetpack blocks

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pcsBup-Fk-p2

## Testing instructions:

- Run `jetpack install plugins/jetpack`.
- Make sure `JETPACK_BLOCKS_BETA` is set to true, to toggle the package code on.
- There should be no functional changes at all compared to the old implementation.

### CODE REVIEW

**Comment by jeherve** [comment]
> This PR adds a new jetpack/forms package and moves the existing contact form code to it.

What would you think about splitting this into multiple parts? We could start by a smaller PR with just the skeleton for the package, and then start adding things to it follow-up PRs. It would be easier to review I think.

**Comment by ice9js** [comment]
>What would you think about splitting this into multiple parts? We could start by a smaller PR with just the skeleton for the package, and then start adding things to it follow-up PRs. It would be easier to review I think.

This was my intention. I was just hoping to get somewhat of a head start here and see how hard it'd actually be to get things to work, as well as maybe get some general feedback on the organization side of things.

File: projects/packages/forms/.phpcs.dir.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<ruleset>
+
+	<rule ref="WordPress.WP.I18n">
+		<properties>
+			<property name="text_domain" type="array">
+				<element value="jetpack-forms" />
```
Comment (CGastrell) [inline projects/packages/forms/.phpcs.dir.xml:7]: Does this mean we start translations from scratch? Should we ask if there's a way to "migrate" translations? Or simply copy the `jetpack` translations domain, then unused entries will start to be removed over time?

File: projects/plugins/jetpack/modules/contact-form.php [hunk 2]
```diff
@@ -18,6 +21,11 @@
  * Additional Search Queries: contact, form, grunion, feedback, submission, contact form, email, feedback, contact form plugin, custom form, custom form plugin, form builder, forms, form maker, survey, contact by jetpack, contact us, forms free
  */
 
+if ( defined( 'JETPACK_BETA_BLOCKS' ) && JETPACK_BETA_BLOCKS ) {
```
Comment (CGastrell) [inline projects/plugins/jetpack/modules/contact-form.php:24]: In my experience this doesn't work on .com, just leaving a note so we don't forget.

See:
- p1669932430568779-slack-CDLH4C1UZ

File: projects/packages/forms/.phpcs.dir.xml (hunk 1 above)
Comment (jeherve) [reply projects/packages/forms/.phpcs.dir.xml:7]: The textdomain change should be automatically handled for us when the package will be required inside the Jetpack plugin.

File: projects/plugins/jetpack/modules/contact-form.php (hunk 2 above)
Comment (jeherve) [reply projects/plugins/jetpack/modules/contact-form.php:24]: This should work on WordPress.com Simple (the flag is set when you're a proxied Automattician), but I would recommend using a specific filter to gate access, one that is only used for forms. This is what Paul mentioned in the thread.

**Comment by ice9js** [comment]
Closing. The work has been split up and merged through a bunch of smaller PRs.

**Pull request closed without merging**